plugin/trino-hive/src/main/java/io/trino/plugin/hive/projection/PartitionProjection.java

```diff
@@ -87,7 +133,19 @@ public Map<String, Optional<Partition>> getProjectedPartitionsByNames(Table tabl
 
     private Partition buildPartitionObject(Table table, String partitionName)
     {
+        List<String> partitionColumns = table.getPartitionColumns().stream()
+                .map(Column::getName)
+                .collect(toImmutableList());
+
         List<String> partitionValues = toPartitionValues(partitionName);
+        ImmutableList.Builder<String> partitionTemplateValues = ImmutableList.builderWithExpectedSize(partitionValues.size());
+        for (int i = 0; i < partitionColumns.size(); i++) {
+            String partitionColumn = partitionColumns.get(i);
+            String partitionValue = columnProjections.get(partitionColumn).toPartitionLocationTemplateValue(partitionValues.get(i));
+            // Don't escape partition value if storage location template is defined
```

Comment:
Nit: expand on this comment to mention that when the template is defined but equivalent to the hive default we still escape.